crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -1702,14 +1726,11 @@ impl Analyzer<'_, '_> {
                             // number.
                             if let Some(types) = self.find_type(&v.enum_name)? {
                                 for ty in types {
-                                    match *ty.normalize() {
-                                        Type::Enum(ref e) => {
-                                            let is_num = !e.has_str;
-                                            if is_num {
-                                                return Ok(());
-                                            }
+                                    if let Type::Enum(ref e) = *ty.normalize() {
+                                        let is_num = e.has_num;
```

Comment:
This is wrong

Reply:
To compare the exact values, rhs must have a name。

But the rhs I found are as follows. at cargo test intersectionReductionStrict__9
Why is the name B empty because it is Tag1.b?
```
EnumVariant(
    EnumVariant {
        span: Span {
            lo: BytePos(
                135,
            ),
            hi: BytePos(
                137,
            ),
            ctxt: #1,
        },
        enum_name: Tag1#1,
        name: None,
        metadata: EnumVariantMetadata {
            common: CommonTypeMetadata {
                implicit: false,
                infected_by_this_in_object_literal: false,
                prevent_converting_to_children: false,
                no_expand: false,
                ignore_no_expand: false,
                contains_infer_type: false,
                prevent_complex_simplification: false,
                resolved_from_var: true,
                prevent_generalization: false,
                prevent_tuple_to_array: false,
            },
        },
    },
)
```

Reply:
I don't know - there's no "context" in types and you have to trace it backward.
Additionally, `has_num` and `has_str` can be used if we can't know the variant name.

Reply:
thx. 

thanks to you, i fixed it